apps/deploy-web/src/components/new-deployment/CreateLease.tsx

```diff
@@ -236,6 +239,44 @@ export const CreateLease: React.FunctionComponent<Props> = ({ dseq }) => {
     setSearch(value);
   };
 
+  const trialProviderCount = useMemo(() => {
+    if (providers) {
+      return providers.filter(provider => {
+        return provider.attributes.some(attribute => {
+          return attribute.key === TRIAL_ATTRIBUTE && attribute.value === "true";
+        });
+      }).length;
+    }
+
+    return 0;
+  }, [providers]);
+
+  const [zeroBidsForTrialWarningDisplayed, setZeroBidsForTrialWarningDisplayed] = useState(false);
+  const { data: block, refetch: getBlock } = useBlock(dseq, {
+    disabled: true
+  });
+
+  useEffect(() => {
+    getBlock();
+  }, [getBlock]);
+
+  useEffect(() => {
+    if (!isTrialing || numberOfRequests === 0 || bids?.length > 0) {
+      setZeroBidsForTrialWarningDisplayed(false);
+      return;
+    }
+
+    const intervalId = setInterval(() => {
+      const blockTime = new Date(block.block.header.time).getTime();
+      const now = new Date().getTime();
+
+      setZeroBidsForTrialWarningDisplayed(now - blockTime > TRIAL_SIGNUP_WARNING_TIMEOUT);
+      clearInterval(intervalId);
+    }, 1000);
+
+    return () => clearInterval(intervalId);
+  }, [block, bids, isTrialing, numberOfRequests]);
```

Comment:
suggestion (non-blocking): I tend to put all the state variables and useEffect's at the top of the file